CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+cmake_minimum_required(VERSION 3.16)
+project(evm_bridge_contracts)
+add_subdirectory (solidity)
+include(ExternalProject)
+# if no cdt root is given use default path
+if(CDT_ROOT STREQUAL "" OR NOT CDT_ROOT)
+   find_package(cdt)
+endif()
+
+ExternalProject_Add(
+   erc2o_project
+   SOURCE_DIR ${CMAKE_SOURCE_DIR}/contracts/erc2o/src
+   BINARY_DIR ${CMAKE_BINARY_DIR}/erc2o
+   CMAKE_ARGS -DCMAKE_TOOLCHAIN_FILE=${CDT_ROOT}/lib/cmake/cdt/CDTWasmToolchain.cmake -DEXTERNAL_DIR=${CMAKE_SOURCE_DIR}/external
+   UPDATE_COMMAND ""
+   PATCH_COMMAND ""
+   TEST_COMMAND ""
+   INSTALL_COMMAND ""
+   BUILD_ALWAYS 1
+   DEPENDS GenerateBytecode
+)
```

Comment:
Something like this should be in the `CMakeLists.txt` of the `antelope_contracts` directory (a rename of the existing `contracts` directory). However, it should use a SOURCE_DIR (and corresponding BINARY_DIR) for the `contracts` sub-directory within `antelope_contracts` and not any specific sub-directory (for each of the specific Antelope contracts) underneath it. The sub-directories should be handled using the `add_subdirectory` CMake command.

Reply:
Refactored folder structure, this should have been resolved.